osquery/utils/aws/aws_util.cpp

```diff
@@ -168,10 +172,133 @@ bool validateIMDSV2RequestAttempts(const char* flagname, std::uint32_t value) {
 
   return true;
 }
+
+std::string awsServiceTypeToString(const AWSServiceType service_type) {
+  switch (service_type) {
+  case AWSServiceType::Kinesis: {
+    return "kinesis";
+  }
+  case AWSServiceType::Firehose: {
+    return "firehose";
+  }
+  case AWSServiceType::STS: {
+    return "sts";
+  }
+  case AWSServiceType::EC2: {
+    return "ec2";
+  }
+  }
+
+  return "unsupported";
+}
+
+Status validateRegion(const std::string& region) {
+  if (!region.empty()) {
+    auto index = kAwsRegions.find(region);
```

Comment:
Is `kAwsRegions` up to date? Are there any reasons for which users might want to use regions that aren't real AWS regions (eg. for S3 in Fleet people sometimes use Minio which will then need a region that's not a real AWS region)?

Reply:
`kAwsRegions` was being used for region validation originally too, so I did not change that.
I did updated it in a different PR though because regions were missing, but it should be unrelated to a non-AWS endpoint.

Though indeed I did not kept the disabling of region validation when the endpoint override is set, because I thought it was unnecessary for non-AWS endpoints, since the url to communicate to is fixed to the endpoint override.
In hindsight it could send it also as metadata to a global endpoint.

I'll put back the check for the endpoint override.

Reply:
Just a heads up that for the STS requests the region **must** be valid; this was an original limitation, that I did not notice in the context of the non-AWS usages.
I did not change the behavior, so this to me would be out of scope; I'm also not exactly sure if there is a need to have custom named regions to provide STS credentials functionality.

The endpoint override was only for Firehose and Kinesis endpoints, not for STS.

Reply:
sgtm, thank you!